src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -262,22 +267,23 @@ public bool NotUsingHttps(string url)
                     waitingTasks.Add(requester.ExecuteAsync(new WaitForRaftIndexCommand(index), context, cts.Token));
                 }
 
-                await Task.WhenAny(waitingTasks);
-                cts.Cancel();
-                foreach (var task in waitingTasks)
+                while (true)
                 {
-                    try
-                    {
-                        await task;
-                    }
-                    catch (OperationCanceledException)
+                    var task = await Task.WhenAny(waitingTasks);
+                    if(task == timeoutTask)
+                        throw new TimeoutException($"Waited too long for the raft command (number {index}) to be executed on any of the relevant nodes to this command.");
+                    if (task.IsCompletedSuccessfully)
```

Comment:
That isn't a good idea, we need to wait until:
* timeout
* _any_ task has completed successfully
* all tasks failed